presidio-analyzer/presidio_analyzer/recognizer_registry_provider.py

```diff
@@ -0,0 +1,156 @@
+        If language wasn't specified, create a recognizer for each supported language.
+
+        :param recognizer: The afordmentioned recognizer.
+        :return: The list of recognizers in the supported languages. 
+        """
+        if isinstance(recognizer, str) or "supported_languages" not in recognizer:
+            return [{"supported_language": language, "context": self._get_recognizer_context(recognizer)} for language in self.supported_languages]
+        if isinstance(recognizer["supported_languages"][0], str):
+            return [{"supported_language": language, "context": None} for language in recognizer["supported_languages"]]
+        return [{"supported_language": language["language"], "context": language["context"]} for language in recognizer["supported_languages"]]
+
+    def _split_recognizers(self, recognizers: Union[Dict[str, Any], str]) -> Tuple[List[Dict[str, Any]],List[Dict[str, Any]]]:
+        """
+        splits the recognizer list to predefined and custom. All recognizers are custom by default though type: 'custom' can be mentioned as well. 
+        This function supports the previous format as well.
+        """
+        predefined =[]
+        custom =[]
+
+        predefined = [recognizer for recognizer in recognizers if isinstance(recognizer, str) or ("type" in recognizer and recognizer["type"] == "predefined")]
+        custom = [recognizer for recognizer in recognizers if not isinstance(recognizer, str) and ("type" not in recognizer or recognizer["type"] == "custom")]
+        return predefined, custom
+
+    def _create_custom_recognizers(self, recognizer: Dict) -> List[EntityRecognizer]:
+        # legacy recognizer
+        if "supported_language" in recognizer:
+            return [PatternRecognizer.from_dict(recognizer)]
+
+        recognizers = []
+
+        for supported_language in self._get_recognizer_languages(recognizer):
+            copied_recognizer = {k: v for k, v in recognizer.items() if k not in ["enabled", "type", "supported_languages"]}
+            recognizers.append(PatternRecognizer.from_dict({**copied_recognizer, **supported_language}))
+
+        return recognizers
+    
+    default_values = {
+        "supported_languages": ["en"],
+        "recognizers": [],
+        "global_regex_flags": re.DOTALL | re.MULTILINE | re.IGNORECASE
+    }
+
+    def create_recognizer_registry(self) -> RecognizerRegistry:
+        fields = {
+            "supported_languages": None,
+            "recognizers": None,
+            "global_regex_flags": None
+        }
+
+        for field in fields.keys():
+            if field not in self.configuration:
+                logger.warning(f'{field} not present in configuration, using default value instead: {self.default_values[field]}')
+            fields[field] = self.configuration.get(field, self.default_values[field])
+
+        self.supported_languages = fields["supported_languages"]
```

Comment:
Does this have to be in self? If yes, please add `self.supported_languages=None` to the init